app_dart/lib/src/service/branch_service.dart

```diff
@@ -134,12 +134,17 @@ class BranchService {
     required gh.RepositorySlug slug,
     required String branchName,
   }) async {
-    return (await githubService.getFileContent(
-      slug,
-      'bin/internal/release-candidate-branch.version',
-      ref: branchName,
-    ))
-        .trim();
+    String branchNameFromFile;
+    try {
+      branchNameFromFile = await githubService.getFileContent(
+        slug,
+        'bin/internal/release-candidate-branch.version',
+        ref: branchName,
+      );
+    } catch (e) {
+      return '$branchName: 404 file not found';
```

Comment:
It would be easier downstream to catch this with an empty string. Then the only check needed is `branch.isEmpty`

Reply:
Synced with Casey and will return empty string, because other people might be able to create a branch with the nonempty name, and gain release branch privilege.